### PR TITLE
Rewrite the osu!catch Ranking Criteria

### PR DESCRIPTION
**Pull request by Darius-Wattimena**
catch RC rewrite/cleanup as per https://osu.ppy.sh/community/forums/topics/1265691

### CODE REVIEW

File: wiki/Ranking_Criteria/osu!catch/en.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # osu!catch ranking criteria
 
+**Note: This document is an extension of the [general ranking criteria](https://osu.ppy.sh/wiki/en/Ranking_Criteria).**
```
Comment (TPGPL) [inline wiki/Ranking_Criteria/osu!catch/en.md]:
```suggestion
*Note: This document is an extension of the [general ranking criteria](/wiki/Ranking_Criteria).*
```

Comment (Darius-Wattimena) [reply wiki/Ranking_Criteria/osu!catch/en.md]: Just pushed a commit with all the things fixed, but thanks for pointing out anyway 👍

File: wiki/Ranking_Criteria/osu!catch/en.md [hunk 2]
```diff
@@ -28,10 +30,20 @@ These terms frequently appear when getting in touch with osu!catch difficulties
...
-- **Higher-snapped dash/hyperdash:** Any dash or hyperdash that doesn't meet the requirement to be a basic one, i.e. the time between the objects is less than the threshold to be classified as basic.
+- **Basic-snapped dash/hyperdash:** Any dash or hyperdash whose time between active objects is at least twice the time required to allow dashes or hyperdashes, respectively. As an example, a hyperdash between objects separated by 250 ms in a Platter classifies as a basic-snapped hyperdash.
+- **Higher-snapped dash/hyperdash:** Any dash or hyperdash that doesn't meet the requirement to be a basic-snapped one, i.e. the time between the objects is less than the threshold to be classified as basic.
 - **Antiflow:** A strong direction or velocity change that goes against a player's natural movement pattern.
 
+#### Snapping reference table
+
+| Difficulty | Basic-snapped dash | Higher-snapped dash | Basic-snapped hyperdash | Higher-snapped hyperdash |
+| --- | --- | --- | --- | --- |
```
Comment (TPGPL) [inline wiki/Ranking_Criteria/osu!catch/en.md:40]:
```suggestion
| :-- | :-- | :-- | :-- | :-- |
```

File: wiki/Ranking_Criteria/osu!catch/en.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # osu!catch ranking criteria
 
+***Note: This document is an extension of the [general ranking criteria](/wiki/Ranking_Criteria).***
```
Comment (TPGPL) [inline wiki/Ranking_Criteria/osu!catch/en.md:3]:
```suggestion
***Notice: This article is an extension of the [general ranking criteria](/wiki/Ranking_Criteria).***
```

**Review by TPGPL — COMMENTED** [review]
some minor formatting changes

File: wiki/Ranking_Criteria/osu!catch/en.md [hunk 4]
```diff
@@ -28,10 +30,20 @@ These terms frequently appear when getting in touch with osu!catch difficulties
...
-- **Basic dash/hyperdash:** Any dash or hyperdash whose time between active objects is at least twice the time required to allow dashes or hyperdashes, respectively. As an example, a hyperdash between objects separated by 250 ms in a Platter classifies as a basic hyperdash.
-- **Higher-snapped dash/hyperdash:** Any dash or hyperdash that doesn't meet the requirement to be a basic one, i.e. the time between the objects is less than the threshold to be classified as basic.
+- **Basic-snapped dash/hyperdash:** Any dash or hyperdash whose time between active objects is at least twice the time required to allow dashes or hyperdashes, respectively. As an example, a hyperdash between objects separated by 250 ms in a Platter classifies as a basic-snapped hyperdash.
+- **Higher-snapped dash/hyperdash:** Any dash or hyperdash that doesn't meet the requirement to be a basic-snapped one, i.e. the time between the objects is less than the threshold to be classified as basic.
```
Comment (MilkyIQ) [inline wiki/Ranking_Criteria/osu!catch/en.md:34]:
```suggestion
- **Higher-snapped dash/hyperdash:** Any dash or hyperdash that doesn't meet the requirement to be a basic-snapped one (i.e. the time between the objects is less than the threshold to be classified as basic).
```
i.e. and e.g. work better in parentheses imo

File: wiki/Ranking_Criteria/osu!catch/en.md [hunk 5]
```diff
@@ -51,10 +63,10 @@ Overall rules and guidelines apply to every kind of osu!catch difficulty. Rhythm
...
 - **Try to have at least one spinner in each difficulty to create variety in the map and fluctuation among scores.** However, if a spinner just doesn't fit anywhere in the song, then there's no need to force one.
-- **Overall Difficulty should have the same value as the Approach Rate.** This is just a standardized value, as Overall Difficulty does not affect gameplay nor the amount of fruits a spinner has, just the maximum score of a difficulty. However, if a difficulty uses a lower Approach Rate than one or more difficulties in the difficulty level below it, then the Overall Difficulty should be equal to that of the highest Overall Difficulty value in the previous difficulty level.
+- **Overall Difficulty should have the same value as the Approach Rate.** This is just a standardized value, as Overall Difficulty does not affect gameplay nor the number of fruits a spinner has, just the maximum score of a difficulty. However, if a difficulty uses a lower Approach Rate than one or more difficulties in the difficulty level below it, then the Overall Difficulty should be equal to that of the highest Overall Difficulty value in the previous difficulty level.
```
Comment (MilkyIQ) [inline wiki/Ranking_Criteria/osu!catch/en.md:66]:
```suggestion
- **Overall Difficulty should have the same value as the Approach Rate.** This is just a standardised value, as Overall Difficulty does not affect gameplay nor does the number of fruits a spinner has; only the maximum score of a difficulty matters. However, if a difficulty uses a lower Approach Rate than one or more difficulties in the difficulty level below it, then the Overall Difficulty should be equal to that of the highest Overall Difficulty value in the previous difficulty level.
```
the wording confuses me on what you're trying to communicate there (second sentence). If my attempt is wrong please do correct

File: wiki/Ranking_Criteria/osu!catch/en.md [hunk 6]
```diff
@@ -119,22 +133,23 @@ Difficulty-specific rules and guidelines do only apply to the difficulty level t
...
-- **Edge dashes must not be used** as they require extremely precise timing which cannot be expected of less-experienced players.
-- **At least 125 ms must be left between circles/sliders and the start of spinners to ensure readability.** As an example, a 1/4 gap would be required at 120 BPM, whereas a 1/3 gap would be required at 160 BPM.
-- **At least 250 ms must be left between circles/sliders and the end of spinners to ensure readability.** As an example, a 1/3 gap would be required at 80 BPM, whereas a 1/2 gap would be required at 120 BPM.
+- **Hyperdashes must have at least a 125 ms gap between their two objects.**
+- **Hyperdashes cannot be used on individual drops and/or slider repetitions.** The accuracy and control required is unreasonable at this level and can create a situation where the player potentially fails to read the slider path.
+- **Hyperdashes of different beat snap must not be used between consecutive fruits.** For example, a 1/2 hyperdash followed by a 1/4 hyperdash.
```
Comment (MilkyIQ) [inline wiki/Ranking_Criteria/osu!catch/en.md:138]:
```suggestion
- **Hyperdashes of differing beat snaps must not be used between consecutive fruits.** For example, a 1/2 hyperdash followed by a 1/4 hyperdash.
```
or maybe?
```suggestion
- **Hyperdashes of a different beat snap must not be used between consecutive fruits.** For example, a 1/2 hyperdash followed by a 1/4 hyperdash.
```

**Review by MilkyIQ — COMMENTED** [review]
Some wording changes

File: wiki/Ranking_Criteria/osu!catch/en.md (hunk 5 above)
Comment (Darius-Wattimena) [reply wiki/Ranking_Criteria/osu!catch/en.md:66]: It is indeed a bit unclear, although your changes would make the last part of the sentence incorrect. I'll rewrite the sentence once all the other suggestions are applied

**Review by MilkyIQ — APPROVED** [review]
LGTM as far as style and grammar go

**Comment by Darius-Wattimena** [comment]
@peppy please check this RC change when you can

**Review by peppy — APPROVED** [review]
Seems fine. Maybe too many "this is to ensure readability" suffixing sentences (aren't all the criteria basically there for this purpose?).

**Pull request merged**